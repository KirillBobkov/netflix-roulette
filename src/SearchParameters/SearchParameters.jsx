import React from 'react';
import { Input, Button } from '../primitives';

export default function SearchArea() {
  return (
    <div className='toolbar__search-parameters'>
        <span className='toolbar__search-description'>Search By</span>
        <Button 
        className='button button--choosen button--left-border'
        text={'Title'} 
        />
        <Button 
        className='button button--right-border'
        text={'Gengre'} 
        />
    </div>
  );
}




