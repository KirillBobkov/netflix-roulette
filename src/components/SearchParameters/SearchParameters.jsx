import React from 'react';
import { Button } from '../primitives';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setSearchBy } from '../../store/actions/filterActions';

class SearchParameters extends React.Component {
  handleSearchBy = (event) => {
    const searchBy = event.target.innerHTML.toLowerCase();
    const { setSearchBy } = this.props;

    setSearchBy(searchBy);
  }

  render() {
    console.log(this.props);
    const { filter: { searchBy } } = this.props;
    const searchByTitleMode = searchBy === 'title';
    const searchByTitleClassName = searchByTitleMode ? 'button--choosen' : '';
    const searchByGenreClassName = !searchByTitleMode ? 'button--choosen' : '';

    return (
      <div className='toolbar__search-parameters'>
        <span className='toolbar__search-description'>Search By</span>
        <Button 
          className={`${searchByTitleClassName} button--left-border`}
          text='Title'
          onClick={this.handleSearchBy}
        />
        <Button 
          className={`${searchByGenreClassName} button--right-border`}
          text='Genres'
          onClick={this.handleSearchBy}
        />
      </div>
      );
  }
}

const mapStateToProps = state => ({ filter: state.filter  });
const mapDispatchToProps = { setSearchBy };
export default connect(mapStateToProps, mapDispatchToProps)(SearchParameters);

SearchParameters.propTypes = {
  setSearchBy: PropTypes.func,
  filter: PropTypes.shape({
    sortBy: PropTypes.string,
    searchBy: PropTypes.string,
    search: PropTypes.string
  })
};




