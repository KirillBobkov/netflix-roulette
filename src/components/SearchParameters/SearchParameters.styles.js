import styled from "styled-components";

export const SearchParametersWrapper = styled.div`
    margin: 0 auto;
    width: 770px;
`;

export const SearchParametersDescription = styled.div`
    display: inline-block;
    margin-right: 29px;
    font-size: 12px;
    text-transform: uppercase;
    color: #ffffff;
`;


        // width: 570px;
        // padding: 14px;
        // background-color: #272727b7;         
