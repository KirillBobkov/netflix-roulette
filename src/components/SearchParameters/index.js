export { default as SearchParameters } from './SearchParametersContainer';
