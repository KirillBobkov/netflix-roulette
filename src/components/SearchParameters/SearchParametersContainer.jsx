import { connect } from 'react-redux';
import { setSearchBy } from '../../store/actions/filterActions';
import { SearchParameters } from './SearchParameters';

const mapStateToProps = state => ({ filter: state.filter  });

const mapDispatchToProps = { setSearchBy };

export default connect(mapStateToProps, mapDispatchToProps)(SearchParameters);