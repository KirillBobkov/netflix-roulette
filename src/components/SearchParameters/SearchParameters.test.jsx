import React from 'react';
import SearchParameters from './SearchParametersContainer';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('SearchParameters component', () => {
    const initialState = {
        filter: {
            sortBy: 'release_date',
            sortOrder: 'asc',
            searchBy: 'title',
            search: '',
        }
    };
      
    const mockStore = configureStore();
    const mountWithStore = component => mount(<Provider store={store}> {component} </Provider>);
    let store, 
        wrapperSearchParameters;

    beforeAll(()=>{
        store = mockStore(initialState);
        wrapperSearchParameters = mountWithStore(<SearchParameters />);
      });


    it('should be render correctly', () => {
        expect(wrapperSearchParameters).toMatchSnapshot();
    });
});