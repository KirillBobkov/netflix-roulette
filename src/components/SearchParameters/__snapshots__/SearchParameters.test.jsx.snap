// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchParameters component should be render correctly 1`] = `ReactWrapper {}`;
